### PR TITLE
mac-os release to homebrew

### PR DESCRIPTION
**Pull request by kacurez**
fixes https://keboola.atlassian.net/browse/SRE-2532
Tato uprava vytvara release mac os signed binarok do homebrew. 
Takze po release sa commitne nova verze homebrew formulae do https://github.com/keboola/homebrew-tap
potom staci nainstalovat:
```
brew tap keboola/homebrew-tap 
brew install keboola-as-code
```
a mal by fungovat command `kbc`.

### Testing builds
Build s celou pipelinou az po commit do https://github.com/keboola/homebrew-tap
https://github.com/keboola/keboola-as-code/actions/runs/1209114165
Netagovany build(preskoci release):
https://github.com/keboola/keboola-as-code/actions/runs/1209144335

### CODE REVIEW

File: .github/workflows/push.yml [hunk 1]
```diff
@@ -170,15 +170,14 @@ jobs:
...
+          brew install coreutils
       - name: Sign the mac binaries with Gon
         env:
           AC_USERNAME: ${{ secrets.AC_USERNAME }}
           AC_PASSWORD: ${{ secrets.AC_PASSWORD }}
         run: |
-          gon -log-level=info -log-json ./gon-amd64.json
-          gon -log-level=info -log-json ./gon-arm64.json
+          gon -log-level=info ./gon-amd64.json
+          gon -log-level=info ./gon-arm64.json
```
Comment (kacurez) [inline .github/workflows/push.yml:180]: dal som prec -log-json a funguje, tj loguje ako human readable string

File: .github/workflows/push.yml [hunk 2]
```diff
@@ -191,3 +190,40 @@ jobs:
...
+        uses: actions/checkout@v2
+      - name: Download Homebrew
+        uses: actions/download-artifact@v2
+        with:
+          name: homebrew
+          path: homebrew
+      - name: Push formulae to keboola/homebrew-tap repository
+        uses: cpina/github-action-push-to-another-repository@main
+        env:
+          API_TOKEN_GITHUB: ${{ secrets.HOMEBREW_RELEASE_GITHUB_PERSONAL_ACCESS_TOKEN }}
```
Comment (kacurez) [inline .github/workflows/push.yml:224]: nato aby som mohol pushnut do https://github.com/keboola/homebrew-tap repo som musel vytvorit personal access token
image

File: homebrew.template.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    sha256 "${ARM_TARGET_SHA256}"
+  end
+  if OS.mac? && Hardware::CPU.intel?
+        url "https://github.com/keboola/keboola-as-code/releases/download/v${TARGET_VERSION}/kbc_${TARGET_VERSION}_darwin_amd64.zip"
+    sha256 "${AMD_TARGET_SHA256}"
+  end
+
+  def install
+    bin.install "kbc"
+    bin.install_symlink Dir["#{libexec}/bin/*"]
```
Comment (kacurez) [inline homebrew.template.rb:18]: brew install keboola-as-code vytvori symlink na `kbc`

File: .github/workflows/push.yml [hunk 4]
```diff
@@ -191,3 +190,40 @@ jobs:
...
+      - name: Upload homebrew formulae to artifacts
+        uses: actions/upload-artifact@v2
+        with:
+          name: homebrew
+          path: homebrew
+
+  release-homebrew:
+    needs:
+      - release-mac-os
+    runs-on: ubuntu-latest
```
Comment (kacurez) [inline .github/workflows/push.yml:211]: musel som to dat do separatneho tasku pretoze na macu sa neda pustit `github-action-push-to-another-repository@main`

**Review by webrouse — COMMENTED** [review]
Par poznamok:
- Spravis ma prosim adminom v  tom `tap` repe, keby nieco.
- Mohli by sme prosim pridat Linux, example: https://github.com/hashicorp/homebrew-tap/blob/0f4e0d343008e407e7c68e53e0c56d1c29cad9dc/Formula/boundary.rb#L22-L34 (moja chyba, ze je to v DOD epicu, ale nie story)
- Inak super, ... este vyskusam ten Linux, ak ho pridas, a da approve

**Review by webrouse — APPROVED** [review]
Toto asi kludne mergnime, a ten Linux mozeme doriesit v dalsom PR.
(ak mas toho vela, kludne si to mozem zobrat)

**Comment by kacurez** [comment]
ok, pridal som ta ako admin do https://github.com/keboola/homebrew-tap/settings/access 
Ten linux mozem pridat v dalsom PR, to imho bude rychle.

**Pull request merged**